src/component/reducer/actions/ToolsActions.ts

```diff
@@ -726,9 +726,9 @@ function levelChangeHandler(draft: Draft<State>, action: LevelChangeAction) {
 
   try {
     for (const spectrum of spectra as Spectrum2D[]) {
-      const contourOptions = spectrum.display.contourOptions;
-      const zoom = contoursManager(spectrum.id, levels, contourOptions);
+      const zoom = contoursManager(spectrum, levels);
       levels[spectrum.id] = zoom.wheel(deltaY, altKey);
+      levels[spectrum.id] = zoom.checkLevel();
```

Comment:
Two lines that assign to the same value means either there's a bug or the first line is useless.